upgrade/util/util.go

```diff
@@ -243,6 +277,55 @@ func CheckUpgradePathSupported(namespace string, lhClient lhclientset.Interface)
 	return nil
 }
 
+// checkEngineUpgradePathSupported returns error if the upgrade path from
+// current EngineImage version to emeta.ControllerAPIVersion is not supported.
+func checkEngineUpgradePathSupported(namespace string, lhClient lhclientset.Interface, newEngineVersion emeta.VersionOutput) (err error) {
```

Comment:
Just noticed in the current implementation, we will periodically check if engine images are compatible or not.

[/controller/engine_image_controller.go#L327-L331](https://github.com/longhorn/longhorn-manager/blob/88d986ee8420fcecc0903e27627063f7518ac69b/controller/engine_image_controller.go#L327-L331)
```go
	if err := engineapi.CheckCLICompatibility(engineImage.Status.CLIAPIVersion, engineImage.Status.CLIAPIMinVersion); err != nil {
		engineImage.Status.Conditions = types.SetCondition(engineImage.Status.Conditions, longhorn.EngineImageConditionTypeReady, longhorn.ConditionStatusFalse, longhorn.EngineImageConditionTypeReadyReasonBinary, "incompatible")
		engineImage.Status.State = longhorn.EngineImageStateIncompatible
		return nil
	}

```

Therefore, the code here is more about runtime check during the upgrade instead of reusing EngineImage::Status, which will be periodically updated by controller reconciling loop.